src/scripts/main.js

```diff
@@ -1,3 +1,27 @@
 'use strict';
 
-// write code here
+document.querySelectorAll('.tree li').forEach((li) => {
+  const childNodes = Array.from(li.childNodes);
+
+  childNodes.forEach((node) => {
+    if (node.nodeType === Node.TEXT_NODE && node.textContent.trim() !== '') {
+      const span = document.createElement('span');
+
+      span.textContent = node.textContent.trim();
+      node.replaceWith(span);
+    }
+  });
+});
+
+document.querySelectorAll('.tree span').forEach((span) => {
+  span.addEventListener('click', (e) => {
+    const parentLi = span.parentElement;
+    const childUl = parentLi.querySelector('ul');
+
+    if (childUl) {
+      childUl.hidden = !childUl.hidden;
+    }
```

Comment:
The toggle functionality is implemented correctly. However, consider adding a CSS class to manage the visibility state instead of directly manipulating the `hidden` attribute for better separation of concerns.